### PR TITLE
Add CI for example_project_bzl_4

### PR DESCRIPTION
**Pull request by patryk-kozak**
Additional CI workflow for building entire `example_project_bzl_4` during CI.

### CODE REVIEW

**Review by borancar — APPROVED** [review]

File: .github/workflows/ci-example-project.yaml [hunk 1]
```diff
@@ -19,6 +19,5 @@ jobs:
         run: ./nix-shell.sh
       - name: Build all
         run: |
-          ./nix-shell.sh "--run \
-          cd example_project_bzl_4 \
-          bazel build //..."
+          ./nix-shell.sh "--run cd example_project_bzl_4" 
+          ./nix-shell.sh "--run bazel build //..."
```
Comment (borancar) [inline .github/workflows/ci-example-project.yaml:23]: Wait, doesn't this lose context now? We're restarting the script, so we're relying on the staying where we were. I think I prefer the set of operations to be atomic within a spawned shell.

Comment (AleksanderGondek) [reply .github/workflows/ci-example-project.yaml]: Seconded.

**Review by borancar — APPROVED** [review]

File: .github/workflows/ci-example-project.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      - name: Checkout code
+        uses:
+          actions/checkout@v2.4.0
+          # Flakes do not like shallow clones
+        with:
+          fetch-depth: 0
+      - name: Install nix-shell
+        run: ./nix-shell.sh
+      - name: Build all
+        run: ./nix-shell.sh --run 'cd example_project_bzl_4 && nix-shell --run "bazel build //..."'
```
Comment (borancar) [inline .github/workflows/ci-example-project.yaml:21]: Why not just `cd example_project_bzl_4 && bazel build //...`? Can we have a comment so nobody wonders in the future?

File: scripts/shell.sh [hunk 3]
```diff
@@ -198,5 +198,5 @@ else
   $NIX_USER_CHROOT_BIN "${NIX_STORE}" bash -c\
     ". ${HOME}/.nix-profile/etc/profile.d/nix.sh\
       && NIX_USER_CONF_FILES=${NIX_EXTRA_CONF_PATH}\
-      nix-shell --pure '${__FILE__}/shell.nix' $@"
+      nix-shell --pure '${__FILE__}/shell.nix'" "$@"
```
Comment (borancar) [inline scripts/shell.sh:201]: Good spot! We definitely need shellcheck :)

File: .github/workflows/ci-example-project.yaml (hunk 2 above)
Comment (AleksanderGondek) [reply .github/workflows/ci-example-project.yaml:23]: Documented and issue raised: [#20](https://github.com/Qarik-Group/disruptor/issues/20)

File: scripts/shell.sh (hunk 3 above)
Comment (AleksanderGondek) [reply scripts/shell.sh:201]: https://github.com/Qarik-Group/disruptor/pull/22/

**Pull request merged**